capsules/src/ble_advertising_driver.rs

```diff
@@ -379,33 +144,53 @@ impl AlarmData {
     }
 }
 
+type AdvPduType = u8;
+
+// BLUETOOTH SPECIFICATION Version 4.2 [Vol 6, Part B], section 2.3.3
+const ADV_IND: AdvPduType = 0b0000;
+#[allow(dead_code)]
+const ADV_DIRECTED_IND: AdvPduType = 0b0001;
+const ADV_NONCONN_IND: AdvPduType = 0b0010;
+#[allow(dead_code)]
+const SCAN_REQ: AdvPduType = 0b0011;
+#[allow(dead_code)]
+const SCAN_RESP: AdvPduType = 0b0100;
+#[allow(dead_code)]
+const CONNECT_IND: AdvPduType = 0b0101;
+const ADV_SCAN_IND: AdvPduType = 0b0110;
+
+/// Process specific memory
 pub struct App {
-    advertisement_buf: Option<kernel::AppSlice<kernel::Shared, u8>>,
-    app_write: Option<kernel::AppSlice<kernel::Shared, u8>>,
-    app_read: Option<kernel::AppSlice<kernel::Shared, u8>>,
-    scan_callback: Option<kernel::Callback>,
-    idx: usize,
     process_status: Option<BLEState>,
-    advertisement_interval_ms: u32,
     alarm_data: AlarmData,
+
+    // Advertising meta-data
+    adv_data: Option<kernel::AppSlice<kernel::Shared, u8>>,
+    address: [u8; PACKET_ADDR_LEN],
+    pdu_type: AdvPduType,
+    advertisement_interval_ms: u32,
     tx_power: u8,
     /// The state of an app-specific pseudo random number.
     ///
     /// For example, it can be used for the pseudo-random `advDelay` parameter.
     /// It should be read using the `random_number` method, which updates it as
     /// well.
     random_nonce: u32,
+
+    // Scanning meta-data
+    scan_buffer: Option<kernel::AppSlice<kernel::Shared, u8>>,
+    scan_callback: Option<kernel::Callback>,
 }
 
 impl Default for App {
     fn default() -> App {
         App {
-            advertisement_buf: None,
             alarm_data: AlarmData::new(),
-            app_write: None,
-            app_read: None,
+            adv_data: None,
+            scan_buffer: None,
+            address: [0; PACKET_ADDR_LEN],
+            pdu_type: ADV_IND,
```

Comment:
Shouldn't the default be `ADV_NONCONN_IND` since that's all Tock supports right now?